tutorials/algorithms/04_vqd.ipynb

```diff
@@ -169,9 +169,10 @@
    "source": [
     "from qiskit.algorithms.eigensolvers import VQD\n",
     "\n",
+    "\n",
     "vqd = VQD(estimator, fidelity, ansatz, optimizer, k=k, betas=betas, callback=callback)\n",
     "result = vqd.compute_eigenvalues(operator = H2_op)\n",
-    "vqd_values = result.optimal_values"
```

Comment:
I did not realize at first why you change that from `optimal_values` to `eigenvalues`. I would have thought that would have been the better field to use as was done before when this was run and checked in here. It seems however that there is an issue, which I raised here, Qiskit/qiskit-terra#10263 where the output from the field is not the values, but the points as can seen in the html which is published out there in the docs.

So this seems a reasonable workaround to have the energy values.

I am though concerned by the lowest energy value - that seems quite a way off (0.021) from where it should be, and evidently was before. The ground state computation for VQD should be doing the same as what VQE does and for H2, a very simple molecule, which is what the SparsePauliOp is the hamiltonian for it should get that answer. Since it has such good values for the 2 excited states, which depend on orthogonality from the ground state, it seems odd. I will note the ansatz shown above is for 3 qubits. Which while that's kinda odd the operator is only 2 qubits so it will get resized by VQD so I would think that confusing - I am not sure why it was done like that - I was looking at the ansatz since that can affect how well the the ground state can be determined.

Reply:
@woodsp-ibm  There is inherently some variation with the VQD values and the VQE's. I am only getting errors after 5 decimal places on my end. I can make the `optimal_values`  patch and then use the optimal_values instead of eigenvalues, removing the numpy.real. I believe the cast to real values from complex points is causing some loss or inaccuracy. Let me know if you agree with my course of action.

Reply:
If that first value is around 1.857 then it would be ok - at 1.837 (rounding it up) its a bit far where it should be was my main concern.

Taking the real part of a complex should not be a problem. 

For this tutorial to use optimal values a bug fix release would have to be done on Terra that included a fix in that regard. If you want to see about fixing the bug as per that issue it should be simple enough - but it maybe better for now to leave it as you have it with eigenvalues since that will work with the current code too.

Reply:
I copied the notebook from your fork and ran it and got `[-1.85727399 -1.24458395 -0.88272262]` which seems fine. I see some small variance between runs - this, I think, is because VQD, by default, chooses a random initial point. Maybe the outcome you committed was really unlucky in chosing an initial point from which it failed to converge well to the minimum. Not sure. I think if you commit something where the outcome is -1.857 something it would be better.

Reply:
It does seem to be the random initial point. If I add this in the VQD cell to seed the rng it uses
```
from qiskit.utils import algorithm_globals
algorithm_globals.random_seed = 220
```
then that value produces the 1.836 each time. Other seed values work eg 210. Though 50 I noticed fails on the excited states and they both end up incorrect and the same.

Bottom line I think its fine as it is. Maybe just commit one where the outcome is the right value. Or you can add that rngs seeding code so anyone running the notebook sees the same thing and does not randomly fail through a poorer initial point starting place.

Sorry, this turned out to be a bit more.... I thought the only issue with the notebook was the deprecated aspect!

Reply:
Its more to do with the initial point as I tried to explain above. I think the anstaz is fine - though arguably it could be confusing that its set to 3 qubits upfront when the operator is a 2 qubit operator. Since the ansatz is a template based circuit VQD can, and does, adjust the number of qubits to match. You can do this `print(result.optimal_circuits[0].decompose())` and print the circuit from the result to see this!

Reply:
Ok I made the anstaz change and it I seems to give me a similar error percentage. I will commit with two qubit anstaz with close values. Let me know if you prefer the three qubit I can revert the changes. 
With regards to your apology, I'm learning a lot so I actually enjoyed the added complexity. Sorry for my limited understanding and thank you for your patience.

Reply:
Changing the ansatz there, for number of qubits, will have no net bearing on the outcome. Internally VQD will try to adjust any ansatz to match the operator https://github.com/Qiskit/qiskit-terra/blob/86df15d3b9d80a74db91590af44af72a050371d2/qiskit/algorithms/eigensolvers/vqd.py#L180 This was done so you could pick say a TwoLocal ansatz and not need to know how many qubits the operator was - which if you specific a higher level problem you may not know at the time. Later the number of qubits can be, and is, set by VQD as it needs to match. My main comment there was more that it was showing a 3 qubit ansatz but it will end up setting it and using it as a 2 qubit one so its kinda misleading to show 3.

Reply:
Can you update this text there too, to say two instead of three

>You can set up, for example, a TwoLocal ansatz with three qubits, and choose SLSQP as the optimization method.